lib/src/view/analysis/tree_view.dart

```diff
@@ -162,124 +131,580 @@ class _InlineTreeViewState extends ConsumerState<AnalysisTreeView> {
+
+      final origin = Offset(-padding, 0);
+
+      final path = Path()..moveTo(origin.dx, origin.dy);
+      path.lineTo(origin.dx, sideLineStartPositions.last.dy);
+      for (final position in sideLineStartPositions) {
+        path.moveTo(origin.dx, position.dy);
+        path.lineTo(origin.dx + padding / 2, position.dy);
       }
+      canvas.drawPath(path, paint);
     }
+  }
 
-    // add the children of the first child
-    widgets.addAll(
-      _buildTreeWidget(
-        pgn,
-        options,
-        parent: firstChild,
-        nodes: firstChild.children,
-        shouldShowAnnotations: shouldShowAnnotations,
-        shouldShowComments: shouldShowComments,
-        inMainline: inMainline,
-        startMainline: false,
-        startSideline: false,
-        initialPath: newPath,
-      ),
+  @override
+  bool shouldRepaint(_IndentPainter oldDelegate) {
+    return oldDelegate.sideLineStartPositions != sideLineStartPositions ||
+        oldDelegate.color != color;
+  }
+}
+
+class _IndentedSideLines extends StatefulWidget {
+  const _IndentedSideLines(
+    this.sideLines, {
+    required this.parent,
+    required this.initialPath,
+    required this.params,
+    required this.nesting,
+  });
+
+  final Iterable<ViewBranch> sideLines;
+
+  final ViewNode parent;
+
+  final UciPath initialPath;
+
+  final _PgnTreeViewParams params;
+
+  final int nesting;
+
+  @override
+  State<_IndentedSideLines> createState() => _IndentedSideLinesState();
+}
+
+class _IndentedSideLinesState extends State<_IndentedSideLines> {
+  late List<GlobalKey> _keys;
+
+  List<Offset> _sideLineStartPositions = [];
+
+  final GlobalKey _columnKey = GlobalKey();
+
+  void _redrawIndents() {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final RenderBox? columnBox =
+          _columnKey.currentContext?.findRenderObject() as RenderBox?;
+      final Offset rowOffset =
+          columnBox?.localToGlobal(Offset.zero) ?? Offset.zero;
+
+      final positions = _keys.map((key) {
+        final context = key.currentContext;
+        final renderBox = context?.findRenderObject() as RenderBox?;
+        final height = renderBox?.size.height ?? 0;
+        final offset = renderBox?.localToGlobal(Offset.zero) ?? Offset.zero;
+        return Offset(offset.dx, offset.dy + height / 2) - rowOffset;
+      }).toList();
+
+      setState(() {
+        _sideLineStartPositions = positions;
+      });
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _redrawIndents();
+  }
+
+  @override
+  void didUpdateWidget(covariant _IndentedSideLines oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.sideLines != widget.sideLines) {
+      _redrawIndents();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final hasHiddenLines = widget.sideLines.any((node) => node.isHidden);
+
+    final visibleSideLines = widget.sideLines.whereNot((node) => node.isHidden);
+
+    _keys = List.generate(
```

Comment:
Doing side effects in `build` is discouraged. Here it looks it can be done in `initState`.

Reply:
Ah, I initially had this line in `initState`, but then the indent lines would get messed up when hiding/collapsing variations - The solution is to update `_keys` in `didUpdateWidget` as well